### PR TITLE
full_impl

### PR DESCRIPTION
**Pull request by YaTut1901**

### CODE REVIEW

File: src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        String query = "SELECT * FROM orders WHERE id = ?";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (!resultSet.next()) {
+                return Optional.empty();
+            }
+            Order order = extractOrderFromResultSet(resultSet);
+            order.setProducts(extractProductsFromOrder(order.getId()));
```
Comment (pavlokhomiak) [inline src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java]: close statement before using another one

File: src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete order with ID = " + id, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Order order) {
+        return deleteById(order.getId());
```
Comment (pavlokhomiak) [inline src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java:124]: is it good practice make so big relations between methods?

File: src/main/java/main/internet/shop/dao/jdbcimpl/ShoppingCartDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        String query = "SELECT * FROM shopping_carts WHERE user_id = ?;";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (!resultSet.next()) {
+                return Optional.empty();
+            }
+            ShoppingCart shoppingCart = extractShoppingCartFromResultSet(resultSet);
+            shoppingCart.setProducts(extractProductsFromShoppingCart(shoppingCart.getId()));
```
Comment (pavlokhomiak) [inline src/main/java/main/internet/shop/dao/jdbcimpl/ShoppingCartDaoJdbcImpl.java]: statement.close()

File: src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create order with id = " + order.getId(), e);
+        }
+        return insertProducts(order);
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders WHERE id = ?";
```
Comment (dmytro-hryhoruk) [inline src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java]: check for deleted?

File: src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        return deleteById(order.getId());
+    }
+
+    private Order extractOrderFromResultSet(ResultSet resultSet) throws SQLException {
+        Order order = new Order(resultSet.getLong("user_id"));
+        order.setId(resultSet.getLong("id"));
+        return order;
+    }
+
+    private List<Product> extractProductsFromOrder(Long id) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java]:
```suggestion
    private List<Product> getOrderProducts(Long id) {
```
You are making a new request to Db

File: src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java]: let's either move this after catch block or the send the same connection  object to method in order not to open connection inside other connection

File: src/main/java/main/internet/shop/dao/jdbcimpl/ShoppingCartDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    private ShoppingCart extractShoppingCartFromResultSet(ResultSet resultSet)
+            throws SQLException {
+        ShoppingCart cart = new ShoppingCart(resultSet.getLong("user_id"));
+        cart.setId(resultSet.getLong("id"));
+        return cart;
+    }
+
+    private List<Product> extractProductsFromShoppingCart(Long cartId) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/main/internet/shop/dao/jdbcimpl/ShoppingCartDaoJdbcImpl.java]: same with naming

File: src/main/java/main/internet/shop/dao/jdbcimpl/ShoppingCartDaoJdbcImpl.java [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        String query = "SELECT * FROM shopping_carts WHERE id = ?;";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (!resultSet.next()) {
+                return Optional.empty();
+            }
+            ShoppingCart shoppingCart = extractShoppingCartFromResultSet(resultSet);
+            shoppingCart.setProducts(extractProductsFromShoppingCart(shoppingCart.getId()));
```
Comment (dmytro-hryhoruk) [inline src/main/java/main/internet/shop/dao/jdbcimpl/ShoppingCartDaoJdbcImpl.java]: same with connection

File: src/main/java/main/internet/shop/dao/jdbcimpl/ShoppingCartDaoJdbcImpl.java [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+                    + shoppingCart.getId(), e);
+        }
+        insertProducts(shoppingCart);
+        return shoppingCart;
+
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String queryForCart = "DELETE FROM shopping_carts WHERE id = ?";
```
Comment (dmytro-hryhoruk) [inline src/main/java/main/internet/shop/dao/jdbcimpl/ShoppingCartDaoJdbcImpl.java]: let's update deleted value but not delete it from DB

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**